Modify unimplemented methods in DPDK SDE to return an error

ffoulkes opened the pull request:
- Modified unimplemented methods in dpdk_sde_wrapper to return ERR_UNIMPLEMENTED instead of OK.

Signed-off-by: Derek G Foster <derek.foster@intel.com>

ffoulkes committed:
Modify unimplemented methods in DPDK SDE to return an error

- Modified unimplemented methods in dpdk_sde_wrapper to return
  ERR_UNIMPLEMENTED instead of OK.

Signed-off-by: Derek G Foster <derek.foster@intel.com>

nupuruttarwar commented on stratum/hal/lib/tdi/dpdk/dpdk_sde_wrapper.cc:
```diff
@@ -241,41 +238,45 @@ ::util::Status TdiSdeWrapper::DeletePort(int device, int port) {
     int device, int port, bool is_in_pps, uint32 burst_size,
     uint64 rate_per_second) {
-  return ::util::OkStatus();
+  return MAKE_ERROR(ERR_UNIMPLEMENTED) << "SetPortShapingRate not implemented";
```
For SetPortShapingRate, EnablePortShaping, SetPortAutonegPolicy, SetPortLoopbackMode, should we return 'ERR_OPER_NOT_SUPPORTED' instead of 'ERR_UNIMPLEMENTED' since these are not applicable for DPDK target.

This brings a generic question, for all functionalities which is different between targets, do we plan to declare a function which returns 'UNSUPPORTED' for functionality not applicable to specific target?

nupuruttarwar replied:
ERE_OPER_NOT_SUPPORTED is defined in stratum/stratum/public/lib/error.cc

ffoulkes committed:
Changed several error returns from unimplemented to unsupported

- Return ERR_OPER_NOT_SUPPORTED for methods the SDK doesn't support
  because they don't apply to the target, and ERR_UNIMPLEMENTED if
  we haven't gotten around to implementing it (or are unsure how or
  whether to do it).

Signed-off-by: Derek G Foster <derek.foster@intel.com>

nupuruttarwar reviewed (approved):
LGTM
